source/fundamentals/indexes.txt

```diff
@@ -0,0 +1,411 @@
+   :end-before: end-single-index
+   :dedent:
+
+The following is an example of a query that is covered by the index
+created in the preceding code snippet:
+
+.. literalinclude:: /includes/fundamentals/code-examples/Indexes.cs
+   :language: csharp
+   :start-after: begin-single-index-query
+   :end-before: end-single-index-query
+   :dedent:
+
+For more information, see :manual:`Single Field Indexes </core/index-single>` in the Server manual.
+
+Compound Indexes
+~~~~~~~~~~~~~~~~
+
+**Compound indexes**  hold references to multiple fields within a collection's documents,
+improving query and sort performance.
+
+The following example creates a compound index on the ``type`` and ``rated`` fields within
+the ``sample_mflix.movies`` collection:
+
+.. literalinclude:: /includes/fundamentals/code-examples/Indexes.cs
+   :language: csharp
+   :start-after: begin-compound-index
+   :end-before: end-compound-index
+   :dedent:
+
+The following is an example of a query that is covered by the index
+created in the preceding code snippet:
+
+.. literalinclude:: /includes/fundamentals/code-examples/Indexes.cs
+   :language: csharp
+   :start-after: begin-compound-index-query
+   :end-before: end-compound-index-query
+   :dedent:
+
+For more information, see :manual:`Compound Indexes </core/index-compound>` in the Server manual.
+
+Multikey Indexes
+~~~~~~~~~~~~~~~~
+
+**Multikey indexes**  are indexes that improve performance for queries that specify a
+field with an index that contains an array value. You can define a multikey index using
+the same syntax as a single field or compound index.
+
+The following example creates a compound, multikey index on the ``rated``, ``genres``
+(an array of Strings), and ``title`` fields within the ``sample_mflix.movies`` collection:
+
+.. literalinclude:: /includes/fundamentals/code-examples/Indexes.cs
+   :language: csharp
+   :start-after: begin-multi-key-index
+   :end-before: end-multi-key-index
+   :dedent:
+
+The following is an example of a query that is covered by the index created in the
+preceding code snippet:
+
+.. literalinclude:: /includes/fundamentals/code-examples/Indexes.cs
+   :language: csharp
+   :start-after: begin-multi-key-query
+   :end-before: end-multi-key-query
+   :dedent:
+
+Multikey indexes behave differently from other indexes in terms of
+query coverage, index bound computation, and sort behavior. To learn more about multikey
+indexes, including a discussion of their behavior and limitations, see the
+:manual:`Multikey Indexes page </core/index-multikey>` in the Server manual.
+
+Clustered Indexes
+~~~~~~~~~~~~~~~~~
+
+**Clustered indexes** instruct a collection to store documents ordered by a key value.
+To create a clustered index, specify the clustered index
+option with the ``_id`` field as the key and the ``Unique`` field as
+``true`` when you create your collection.
+
+The following example creates a clustered index on the ``_id`` field in
+the ``tea.vendors`` collection:
```

Comment:
To be more inline with the other examples on this page, can we make this example create a new collection in the sample_mflix database instead of a new tea database? I also think it could be reworded a bit:
```suggestion
The following example creates a clustered index on the ``_id`` field while creating a new ``tea.vendors`` collection:
```